### PR TITLE
Feature #38443 : Stock location barcode - Fix barcode generation checking rule

### PR DESCRIPTION
**Pull request by abo-axelor**

### CODE REVIEW

File: axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java [hunk 1]
```diff
@@ -64,32 +64,35 @@ public StockLocation save(StockLocation stockLocation) {
...
+          boolean addPadding = false;
+          InputStream inStream =
+              barcodeGeneratorService.createBarCode(
+                  stockLocation.getSerialNumber(), barcodeTypeConfig, addPadding);
+          if (inStream != null) {
+            MetaFile barcodeFile =
+                metaFiles.upload(
+                    inStream, String.format("StockLocationBarCode%d.png", stockLocation.getId()));
+            stockLocation.setBarCode(barcodeFile);
+          }
```
Comment (ale-axelor) [inline axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java:88]: This block of code is duplicated in ProductBaseRepository, OperationOrderService and TrackingNumberManagementRepository, so please create a new method in barcodeGeneratorService to create and upload the barcode file

**Review by ale-axelor — CHANGES_REQUESTED** [review]

File: axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java [hunk 2]
```diff
@@ -64,32 +64,35 @@ public StockLocation save(StockLocation stockLocation) {
...
-                stockLocation.getSerialNumber(), barcodeTypeConfig, addPadding);
-        if (inStream != null) {
-          MetaFile barcodeFile =
-              metaFiles.upload(
-                  inStream, String.format("StockLocationBarCode%d.png", stockLocation.getId()));
-          stockLocation.setBarCode(barcodeFile);
+    if (appStock != null && appStock.getActivateStockLocationBarCodeGeneration()) {
+
+      BarcodeTypeConfig barcodeTypeConfig = stockLocation.getBarcodeTypeConfig();
+      // Apply defaulting if needed
```
[the remark this answers is not in the document]
Comment (abo-axelor) [reply axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java]: Done

File: axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java [hunk 3]
```diff
@@ -64,32 +64,35 @@ public StockLocation save(StockLocation stockLocation) {
...
+      if (appStock.getEditStockLocationBarcodeType() == false) {
```
[the remark this answers is not in the document]
Comment (abo-axelor) [reply axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java]: Done

File: axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java (hunk 1 above)
Comment (abo-axelor) [reply axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/StockLocationStockRepository.java]: Done. I have implemented a new method on BarcodeGeneratorService and have updated all impacted code.

**Pull request merged**